--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -507,6 +507,8 @@ m68*-*-netbsd*)	targ_emul=m68kelfnbsd
 			;;
 m68*-*-haiku*)		targ_emul=m68kelf
 			;;
+m68*-*-obos*)           targ_emul=m68kelfnbsd
+                        ;;
 m68*-*-*)		targ_emul=m68kelf
 			;;
 mcore-*-pe)		targ_emul=mcorepe ;
@@ -1020,6 +1022,10 @@ x86_64-*-linux-*)	targ_emul=elf_x86_64
 			targ_extra_libpath="elf_i386 elf32_x86_64"
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'`
 			;;
+x86_64-*-obos*)
+			targ_emul=elf_x86_64
+			targ_extra_emuls="elf_x86_64 elf_i386"
+			;;
 x86_64-*-redox*)	targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386
 			;;
