diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c6f6cc..deaec79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -954,7 +954,7 @@ if(SDLMIXER_MIDI_NATIVE)
         enable_language(CXX)
         target_sources(${sdl3_mixer_target_name} PRIVATE src/codecs/native_midi/native_midi_haiku.cpp ${midi_common_sources})
         target_link_libraries(${sdl3_mixer_target_name} PRIVATE midi)
-    elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
+    elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux" AND NOT DEFINED WE_ARE_REALLY_OBOS)
         list(APPEND SDLMIXER_MIDI_NATIVE_ENABLED TRUE)
         target_sources(${sdl3_mixer_target_name} PRIVATE src/codecs/native_midi/native_midi_linux_alsa.c ${midi_common_sources})
         target_link_libraries(${sdl3_mixer_target_name} PRIVATE asound)
