--- a/os/access.c
+++ b/os/access.c
@@ -1852,7 +1852,7 @@ siHostnameAddrMatch(int family, void *addr, int len,
         if ((hp = _XGethostbyname(hostname, hparams)) != NULL) {
 #ifdef h_addr                   /* new 4.3bsd version of gethostent */
             /* iterate over the addresses */
-            for (addrlist = hp->h_addr_list; *addrlist; addrlist++)
+            for (addrlist = (const char**)hp->h_addr_list; *addrlist; addrlist++)
 #else
             addrlist = &hp->h_addr;
 #endif
