--- a/hw/xfree86/os-support/xf86_OSlib.h
+++ b/hw/xfree86/os-support/xf86_OSlib.h
@@ -176,7 +176,7 @@ extern _X_HIDDEN char xf86SolarisFbDev[PATH_MAX];
 /**************************************************************************/
 /* Linux or Glibc-based system                                            */
 /**************************************************************************/
-#if defined(__linux__) || defined(__GLIBC__) || defined(__CYGWIN__)
+#if defined(__linux__) || defined(__GLIBC__) || defined(__CYGWIN__) || defined(__obos__)
 #include <sys/ioctl.h>
 #include <signal.h>
 #include <stdlib.h>