diff --git a/home/oberrow/bash-5.2.37/support/config.sub b/home/oberrow/Code/bash-5.2.37/support/config.sub
index 9b62e37..91d08bd 100644
--- a/home/oberrow/bash-5.2.37/support/config.sub
+++ b/home/oberrow/Code/bash-5.2.37/support/config.sub
@@ -1723,7 +1723,7 @@ case $os in
 	# Now accept the basic system types.
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
-	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | obos* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
