diff --git a/home/oberrow/bash-5.2.37/lib/termcap/termcap.c b/home/oberrow/Code/bash-5.2.37/lib/termcap/termcap.c
index 87fae05..7c71660 100644
--- a/home/oberrow/bash-5.2.37/lib/termcap/termcap.c
+++ b/home/oberrow/Code/bash-5.2.37/lib/termcap/termcap.c
@@ -295,7 +295,7 @@ tgetst1 (ptr, area)
 short ospeed;
 /* If OSPEED is 0, we use this as the actual baud rate.  */
 int tputs_baud_rate;
-__private_extern__ char PC = '\0';
+ __private_extern__ char PC = '\0';
 
 /* Actual baud rate if positive;
    - baud rate / 100 if negative.  */
