diff --git a/home/oberrow/bash-5.2.37/lib/termcap/tparam.c b/home/oberrow/Code/bash-5.2.37/lib/termcap/tparam.c
index c4bff08..d43df32 100644
--- a/home/oberrow/bash-5.2.37/lib/termcap/tparam.c
+++ b/home/oberrow/Code/bash-5.2.37/lib/termcap/tparam.c
@@ -61,6 +61,7 @@ char *realloc ();
 #endif
 
 #ifndef emacs
+#include <unistd.h>
 static void
 memory_out ()
 {
@@ -123,8 +124,8 @@ tparam (string, outstring, len, arg0, arg1, arg2, arg3)
   return tparam1 (string, outstring, len, NULL, NULL, arg);
 }
 
-__private_extern__ char *BC;
-__private_extern__ char *UP;
+ __private_extern__ char *BC;
+ __private_extern__ char *UP;
 
 static char tgoto_buf[50];
 
