--- a/configure
+++ b/configure
@@ -4052,6 +4052,9 @@ then
 	*-*-linux*)
 		ac_sys_system=Linux
 		;;
+	*-*-obos*)
+		ac_sys_system=Linux
+		;;
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
@@ -4500,6 +4503,15 @@ if test "$cross_compiling" = yes; then
 			_host_ident=$host_cpu
 		esac
 		;;
+	*-*-obos*)
+		case "$host_cpu" in
+		arm*)
+			_host_ident=arm
+			;;
+		*)
+			_host_ident=$host_cpu
+		esac
+		;;
 	*-*-cygwin*)
 		_host_ident=
 		;;
